service/history/execution/mutable_state_builder.go

```diff
@@ -2659,6 +2659,28 @@ func (e *mutableStateBuilder) AddFailWorkflowEvent(
 	return event, nil
 }
 
+func (e *mutableStateBuilder) AddCancelWorkflowEvent(
```

Comment:
we don't have this function before?

Reply:
Not that I saw anyway.  A bit surprisingly.
I can go search more comprehensively though.

Reply:
we have `AddWorkflowExecutionCanceledEvent`, just noticed that you added this but didn't add `ReplicateWorkflowExecutionCanceledEvent `

Reply:
ah hah.  yeah, that should've been a sign 🤔 

thanks + @davidporter-id-au , good catches :)